lib/resque/scheduler.rb

```diff
@@ -175,19 +175,21 @@ def env_matches?(configured_env)
       # Handles queueing delayed items
       # at_time - Time to start scheduling items (default: now).
       def handle_delayed_items(at_time = nil)
-        timestamp = Resque.next_delayed_timestamp(at_time)
-        if timestamp
-          procline 'Processing Delayed Items'
-          until timestamp.nil?
-            enqueue_delayed_items_for_timestamp(timestamp)
-            timestamp = Resque.next_delayed_timestamp(at_time)
+        at_time ||= Time.now
+        item = Resque.next_delayed_item(before: at_time)
+        procline 'Processing Delayed Items'
+
+        until item.nil?
+          handle_shutdown do
+            if master?
+              enqueue_delayed_item(item)
```

Comment:
Can we inline this and remove the `enqueue_delayed_item` method? All it's doing is logging the item and calling `enqueue`

Reply:
:+1: